pkg/reconciler/pipelinerun/resources/pipelineref_test.go

```diff
@@ -81,3 +88,82 @@ func TestPipelineRef(t *testing.T) {
 		})
 	}
 }
+
+func TestGetPipelineFunc(t *testing.T) {
+	// Set up a fake registry to push an image to.
+	s := httptest.NewServer(registry.New())
+	defer s.Close()
+	u, err := url.Parse(s.URL)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	testcases := []struct {
+		name            string
+		localPipelines  []runtime.Object
+		remotePipelines []runtime.Object
+		ref             *v1beta1.PipelineRef
+		expected        runtime.Object
+	}{{
+		name: "remote-pipeline",
+		localPipelines: []runtime.Object{
+			tb.Pipeline("simple", tb.PipelineType, tb.PipelineNamespace("default"), tb.PipelineSpec(tb.PipelineTask("something", "something"))),
+			tb.Pipeline("dummy", tb.PipelineType, tb.PipelineNamespace("default")),
+		},
+		remotePipelines: []runtime.Object{
+			tb.Pipeline("simple", tb.PipelineType, tb.PipelineNamespace("default")),
+			tb.Pipeline("dummy", tb.PipelineType, tb.PipelineNamespace("default")),
+		},
```

Comment:
Those two do not change in the test cases, we can extract term as variable and reduce the test struct.